src/crates-io/lib.rs

```diff
@@ -170,11 +171,12 @@ impl Registry {
         Ok(())
     }
 
-    pub fn search(&mut self, query: &str, limit: u8) -> Result<Vec<Crate>> {
+    pub fn search(&mut self, query: &str, limit: u8) -> Result<(Vec<Crate>, u32)> {
         let body = try!(self.req(format!("/crates?q={}&per_page={}", query, limit), None, Get,
                                  Auth::Unauthorized));
 
-        Ok(json::decode::<Crates>(&body).unwrap().crates)
+        let crates = json::decode::<Crates>(&body).unwrap();
```

Comment:
I have used the same style of the rest of the crates-io library. On the other hand, the try! can't be used directly because the json::decode DecodeError can be directly converted to Error.